jib-core/src/main/java/com/google/cloud/tools/jib/http/Connection.java

```diff
@@ -87,41 +88,32 @@
    * Registers proxy credentials onto transport client, in order to deal with proxies that require
    * basic authentication.
    *
-   * @param transport
+   * @param transport Apache HTTP transport
    */
-  private static void addProxyCredentials(HttpTransport transport) {
-    DefaultHttpClient httpClient =
-        (DefaultHttpClient) ((ApacheHttpTransport) transport).getHttpClient();
-
-    boolean httpProxy = System.getProperty("http.proxyHost") != null;
-    boolean httpCreds =
-        System.getProperty("http.proxyUser") != null
-            && System.getProperty("http.proxyPassword") != null;
-    if (httpProxy && httpCreds) {
-      httpClient
-          .getCredentialsProvider()
-          .setCredentials(
-              new AuthScope(
-                  System.getProperty("http.proxyHost"),
-                  Integer.parseInt(System.getProperty("http.proxyPort", "8080"))),
-              new UsernamePasswordCredentials(
-                  System.getProperty("http.proxyUser"), System.getProperty("http.proxyPassword")));
-    }
-    boolean httpsProxy = System.getProperty("https.proxyHost") != null;
-    boolean httpsCreds =
-        System.getProperty("https.proxyUser") != null
-            && System.getProperty("https.proxyPassword") != null;
-    if (httpsProxy && httpsCreds) {
-      httpClient
-          .getCredentialsProvider()
-          .setCredentials(
-              new AuthScope(
-                  System.getProperty("https.proxyHost"),
-                  Integer.parseInt(System.getProperty("https.proxyPort", "443"))),
-              new UsernamePasswordCredentials(
-                  System.getProperty("https.proxyUser"),
-                  System.getProperty("https.proxyPassword")));
+  private static void addProxyCredentials(ApacheHttpTransport transport) {
+    addProxyCredentials(transport, "https");
+    addProxyCredentials(transport, "http");
+  }
+
+  private static void addProxyCredentials(ApacheHttpTransport transport, String protocol) {
+    Preconditions.checkArgument(protocol.equals("http") || protocol.equals("https"));
+
+    String proxyHost = System.getProperty(protocol + ".proxyHost");
+    String proxyUser = System.getProperty(protocol + ".proxyUser");
+    String proxyPassword = System.getProperty(protocol + ".proxyPassword");
+    if (proxyHost == null || proxyUser == null || proxyPassword == null) {
+      return;
     }
+
+    String defaultProxyPort = protocol.equals("http") ? "80" : "443";
+    int proxyPort = Integer.parseInt(System.getProperty(protocol + ".proxyPort", defaultProxyPort));
```

Comment:
Should we catch the `NumberFormatException`?

Reply:
In principle, I think `addProxyCredentials()` and hence `getConnectionFactory()` should throw some custom checked exception as the error is due to an external input, so that a higher-level application can act accordingly if this happens, but from a practical standpoint, I don't feel it is worth disrupting a large part of the codebase for the sake of propagating a checked exception.

So, I'm leaning toward of letting it throw the runtime exception as-is, meaning that if any application (including our Maven/Gradle plugins) ever surfaces the `NumberFormatException`, then it would indicate, righteously, a genuine program bug of the application. That said, our plugins should never let this happen. One way we can achieve this is to validate `http(s).proxyPort` at the higher application level. In fact, we are already doing such a check with `jib.httpTimeout`. I'll add a check in `JibSystemProperties` for `http(s).proxyPort`, and I think it's the best option here.